docs/proposals/odo-v3-cli/odo-delete-component.md

```diff
@@ -0,0 +1,17 @@
+# `odo delete component`
+Delete component.
+By default it deletes component from cluster (deletes all resources from both `odo dev` and `odo deploy` that were crated on the cluster that belong to the component) that in the current directory.
+
+Similarly to `odo describe component` this commands works in two modes.
```

Comment:
Why do we want multiple modes with everything? What's the obsession here to provide more ways/modes with every command? Why can't we figure just ONE way of doing things, doing it well without breaking, and then if user feedback is to have more ways, we can have a discussion to add more!

Reply:
The idea is that we want to make sure that odo can be useful for users that don't want to use "Devfile" yet. 

This is something that we discussed for a long time. And If I remember you agreed with this.

Reply:
> The idea is that we want to make sure that odo can be useful for users that don't want to use "Devfile" yet.

But if they already have a component that they want to delete, aren't they using devfile already?

> This is something that we discussed for a long time. And If I remember you agreed with this.

Yes, I remember agreeing to it. But my concern right now is to keep odo slim and less bloated, and add features only when "real" users ask for something. That makes development faster, and maintenance simpler.

Do you truly feel that odo is ready to support a scenario where devfile.yaml doesn't exist, and each "create" command would create a manifest file and then `odo push`/`odo dev` would create those resources on the cluster? I don't think we have discussed that sort of flow in enough detail to start working on it for v3alpha1.

Also, I'll repeat the same question/point that I have been annoying you with since a long time - did any "real" user(s) request for this or is it just our imagination that some users would like to have this?

Reply:
> But if they already have a component that they want to delete, aren't they using devfile already?

in case of `odo describe component`, the component will probably be always created based on the devfile, but that devfile might not be available to odo.


> Also, I'll repeat the same question/point that I have been annoying you with since a long time - did any "real" user(s) request for this or is it just our imagination that some users would like to have this?

Yes, this is based on the feedback from users.